apps/system/test/marionette/update_addon_test.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var AppInstall = require('./lib/app_install');
+var NotificationList = require('./lib/notification.js').NotificationList;
+var createAppServer = require(
+  '../../../../apps/homescreen/test/marionette/server/parent');
+
+marionette('Update add-on', function() {
+
+  var client = marionette.client({
+    profile: {
+      prefs: {
+        'dom.apps.developer_mode': true
+      },
+      settings: {
+        'addons.auto_update': true,
+        'addons.update_notify': true
+      }
+    },
+    desiredCapabilities: { raisesAccessibilityExceptions: true }
+  });
+
+  var appInstall, server, system, notificationList;
+
+  suiteSetup(function(done) {
+    var app = __dirname + '/fixtures/template_addon';
+    createAppServer(app, client, function(err, _server) {
+      server = _server;
+      done(err);
+    });
+  });
+
+  suiteTeardown(function(done) {
+    server.close(done);
+  });
+
+  setup(function() {
+    appInstall = new AppInstall(client);
+    notificationList = new NotificationList(client);
+    system = client.loader.getAppClass('system');
+    system.waitForFullyLoaded();
+    client.switchToFrame();
+  });
+
+  test('Basic add-on update test', function() {
+
+    appInstall.installPackage(server.webappUpdateURL);
+
+    // Update the manifest.
+    server.setRoot(__dirname + '/fixtures/template_addon_updated');
+
+    // Check for an update.
+    appInstall.stageUpdate(server.webappUpdateURL);
+
+    // We should get a notification on the first time an
+    // add-on is automatically installed
+    notificationList.waitForNotificationCount(2);
+  });
+
+
+  test('Auto update disabled test', function() {
+
+    client.settings.set('addons.auto_update', false);
+
+    appInstall.installPackage(server.webappUpdateURL);
+
+    // Update the manifest.
+    server.setRoot(__dirname + '/fixtures/template_addon_updated');
+
+    // Check for an update.
+    appInstall.stageUpdate(server.webappUpdateURL);
+
+    // Dont wait 30 seconds for the notification to show
+    system.sendSystemUpdateNotification();
+
+    notificationList.waitForNotificationCount(1);
```

Comment:
Like above, it would be more robust to make sure none of the notifications are update notifications for add-ons.